addons/mail/models/mail_template.py

```diff
@@ -218,16 +227,19 @@ def _generate_template(self, res_ids, render_fields):
                     values['scheduled_date'] = parsed_datetime.replace(tzinfo=None) if parsed_datetime else False
 
                 # technical settings
-                values.update(
-                    mail_server_id=template.mail_server_id.id or False,
-                    auto_delete=template.auto_delete,
-                    model=template.model,
-                    res_id=res_id or False,
-                    attachment_ids=[attach.id for attach in template.attachment_ids],
-                )
-
-            # Add report in attachments: generate once for all template_res_ids
-            if template.report_template:
+                if 'attachments' in render_fields or 'attachment_ids' in render_fields:
+                    values['attachment_ids'] = [attach.id for attach in template.attachment_ids]
```

Comment:
while you change this
```suggestion
                    values['attachment_ids'] = template.attachment_ids.ids
```